squirrel.frontier/src/main/java/org/dice_research/squirrel/components/FrontierComponent.java

```diff
@@ -219,17 +216,20 @@ public void handleData(byte[] data, ResponseHandler handler, String responseQueu
     }
 
     private void responseToUriSetRequest(ResponseHandler handler, String responseQueueName, String correlId,
-            UriSetRequest uriSetRequest) {
+                                         UriSetRequest uriSetRequest) {
         if (handler != null) {
             // get next UriSet
             try {
                 List<CrawleableUri> uris = frontier.getNextUris();
                 LOGGER.trace("Responding with a list of {} uris.",
-                        uris == null ? "null" : Integer.toString(uris.size()));
+                    uris == null ? "null" : Integer.toString(uris.size()));
                 handler.sendResponse(serializer.serialize(new UriSet(uris)), responseQueueName, correlId);
                 if (uris != null && uris.size() > 0) {
-                    workerGuard.putUrisForWorker(uriSetRequest.getWorkerId(), uriSetRequest.workerSendsAliveMessages(),
-                            uris);
+                    hasUrisToCrawl.put(uriSetRequest.getWorkerId(), true);
+                    workerGuard.putUrisForWorker(uriSetRequest.getWorkerId(),
+                        uriSetRequest.workerSendsAliveMessages(), uris);
+                } else {
+                    hasUrisToCrawl.put(uriSetRequest.getWorkerId(), false);
```

Comment:
This looks like you are re-introducing old, deprecated code. Please double check this.